bots/push-rewrite

```diff
@@ -30,6 +30,14 @@ def execute(*args):
 def git(*args):
     return execute("git", *args).strip()
 
+def find_pr_with_sha(api, sha):
+    pulls = api.pulls()
+    for pull in pulls:
+        if pull["head"]["sha"] == sha:
+            return pull
+    sys.stderr.write("Could not find pull with revision {0}.\n".format(sha))
+    return 1
```

Comment:
This will just be interpreted by the  caller as "pull request #1", so please rather do sys.exit() here.